### PR TITLE
Update docs with more information on default Delayed scheduler

### PR DESCRIPTION
**Pull request by guillaumeeb**
Just a small documentation update in order to find more easily what is the default scheduler for delayed API.

### CODE REVIEW

**Comment by guillaumeeb** [comment]
cc @ncclementi

File: docs/source/delayed.rst [hunk 1]
```diff
@@ -209,3 +209,14 @@ on the side-effect of another task. In these cases you can use
 ``sum_data`` will operate on DATA only after both the expected items have
 been appended to it. ``bind`` can also be used along with direct dependencies
 passed through the function arguments.
+
+
+Execution
+---------
+
+By default, Dask Delayed uses the threaded scheduler in order to avoid data
+transfer costs.
```
Comment (ncclementi) [inline docs/source/delayed.rst:218]: Do we know this is the reason why? Asking because I do not know...

File: docs/source/delayed.rst [hunk 2]
```diff
@@ -209,3 +209,14 @@ on the side-effect of another task. In these cases you can use
...
 passed through the function arguments.
+
+
+Execution
+---------
+
+By default, Dask Delayed uses the threaded scheduler in order to avoid data
+transfer costs.  
+You should consider using multiprocessing or `dask.distributed`_ scheduler 
+on a local machine or on a cluster if your code does not release the GIL well.
```
Comment (ncclementi) [inline docs/source/delayed.rst:220]: I wonder if we can add a link to some docs that explain what this mean, I only found this but not sure if it's enough https://docs.dask.org/en/stable/graphs.html#avoid-holding-the-gil

**Review by ncclementi — COMMENTED** [review]
Thanks @guillaumeeb, adding this context is helpful. I left some comments/questions. 

It might be helpful having someone else to take a look at this too just in case, maybe @jacobtomlinson

File: docs/source/10-minutes-to-dask.rst [hunk 3]
```diff
@@ -490,8 +490,12 @@ Scheduling
 After you have generated a task graph, it is the scheduler's job to execute it
 (see :doc:`scheduling`).
 
-By default when you call ``compute`` on a Dask object, Dask uses the thread
-pool on your computer to run computations in parallel.
+By default, for the majority of Dask APIs, when you call ``compute`` on a Dask object, 
+Dask uses the threaded scheduler, so basically a thread pool on your computer
+to run computations in parallel.
```
[the remark this answers is not in the document]
Comment (guillaumeeb) [reply docs/source/10-minutes-to-dask.rst:495]: Yep, this sounds better!

File: docs/source/delayed.rst (hunk 1 above)
Comment (guillaumeeb) [reply docs/source/delayed.rst:218]: There you got me, I have to admit I just copy/pasted this from Dask Array documentation. I also know this is true, avoiding inter process data exchange can surely optimize a computation.

But in the end, I'm not sure if this is the real reason why Delayed uses the threaded scheduler by default.

File: docs/source/delayed.rst (hunk 2 above)
Comment (guillaumeeb) [reply docs/source/delayed.rst:220]: Here again, I only copy pasted something. I just know that multi-threading with pure Python code or libraries that hold to the GIL won't work. 

There are some mention of the GIL here too: https://docs.dask.org/en/stable/scheduling.html#local-processes.

**Review by jacobtomlinson — APPROVED** [review]
This looks reasonable to me. I agree with @ncclementi that more links are better. But if there isn't anything obvious to link to then I suggest we go ahead and merge.

**Comment by ncclementi** [comment]
@guillaumeeb If you want to add the links we mentioned above, that would be great. After that, we can go ahead and merge.

**Review by ncclementi — APPROVED** [review]
LGTM thanks @guillaumeeb

**Comment by guillaumeeb** [comment]
Thanks to both of you!

**Pull request merged**